src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -57,28 +57,48 @@ class FileProcessor(
     } yield parsedJson
   }
 
+  def parseUri(potentialUri: Option[String]): IO[Option[ParsedUri]] = {
+    potentialUri.map { uri =>
+      val citeRegex = "^.*/id/([a-z]*)/".r
```

Comment:
1.  `id/blah` would return a cite of blah which is fine. It would fail later on when it tries to get the trimmed string because it's expecting a year but that's fine. TRE have given us the spec for the urls and this matches.
2. Two slashes would match and give a cite of empty string which would fail the department and series lookup which is what we want.